python-avd/pyavd/_eos_designs/shared_utils/underlay.py

```diff
@@ -110,3 +111,14 @@ def underlay_ipv6_numbered(self: SharedUtilsProtocol) -> bool:
                 )
                 raise AristaAvdInvalidInputsError(msg)
         return self.inputs.underlay_ipv6_numbered
+
+    @cached_property
+    def underlay_isis_authentication_cleartext_key(self: SharedUtilsProtocol) -> str | None:
+        """Returns the encrypted key for underlay isis authentication."""
+        if self.inputs.underlay_isis_authentication_cleartext_key is not None:
+            return isis_encrypt(
+                password=cast("str", self.inputs.underlay_isis_authentication_cleartext_key),
+                mode=cast("str", self.inputs.underlay_isis_authentication_mode),
+                key=cast("str", self.isis_instance_name),
+            )
+        return None
```

Comment:
Why do you need the casts?
```suggestion
        if self.inputs.underlay_isis_authentication_cleartext_key is None:
            return None
        return isis_encrypt(
            password=cast("str", self.inputs.underlay_isis_authentication_cleartext_key),
            mode=cast("str", self.inputs.underlay_isis_authentication_mode),
            key=cast("str", self.isis_instance_name),
        )
```

Reply:
mode argument in isis_encrypt is str and self.inputs.underlay_isis_authentication_mode is str or None. So it shows type hint error.